actionmailer/test/log_subscriber_test.rb

```diff
@@ -35,6 +41,29 @@ def test_deliver_is_notified
     BaseMailer.deliveries.clear
   end
 
+  def test_deliver_is_notified_when_exception_is_raised
+    @original_delivery_method = ActionMailer::Base.delivery_method
+    ActionMailer::Base.delivery_method = ExceptionDelivery
+
+    mail = BaseMailer.welcome
+    assert_raise Net::SMTPAuthenticationError do
+      mail.deliver_now
+    end
+    wait
+
+    assert_equal(0, @logger.logged(:info).size)
+
+    assert_equal(1, @logger.logged(:error).size)
+    assert_match(/Net::SMTPAuthenticationError was raised when sending mail to system@test.lindsaar.net/, @logger.logged(:error).first)
```

Comment:
Thanks for the contribution Eileen! It needs to be approved by someone to be merged in the master branch of rails?
To use it now I would have to fork rails in my machine and copy this code, right?
I want to be able to contribute too. I've read a post of yours about it. Any other advice on it would be very appreciated! :D